istioctl/pkg/waypoint/waypoint.go

```diff
@@ -43,13 +50,95 @@ import (
 )
 
 var (
-	revision = ""
-
+	revision      = ""
+	waitReady     bool
+	waitTimeout   time.Duration
 	allNamespaces bool
+	revisionLabel = "istio.io/rev"
 
 	deleteAll bool
 )
 
+func isWaypointDeploymentReady(ctx cli.Context, kubeClient kube.CLIClient, gw *gateway.Gateway) (bool, error) {
```

Comment:
We should check the Gateway is ready IMO, not the Deployment. The Gateway is even higher abstraction than Deployment

Reply:
+1 for the Gateway. The command deploys only a Gateway resource, and should check that instead of the Deployment.

Reply:
Thank you for pointing it out, I was not aware of the Gateway conditions.
I presume the one I should check is the "Programmed" one which from my understanding is to true when the initial configuration has been generated, sent and acked by the data-plane.

I have now updated the code to check on the Gateway object.